src/lib.rs

```diff
@@ -301,6 +418,16 @@ impl Contract {
         BlockHeightCallbackRetValue { proposal_id: proposal.id }
     }
 
+    pub fn set_rfp_block_height_callback(
+        &mut self,
+        #[allow(unused_mut)] mut rfp: RFP,
+        #[callback_unwrap] set_result: SetReturnType,
+    ) -> BlockHeightCallbackRetValue {
+        rfp.social_db_post_block_height = set_result.block_height.into();
+        self.rfps.push(&rfp.clone().into());
```

Comment:
Instead of cloning the whole thing, record the `porposal_id`, so you don't need the whole `rfp` object anymore, so you can just `rfp.into()` it